ROADMAP.md

```diff
@@ -0,0 +1,117 @@
+* Despite working on Swift projects, the majority of our iOS developers are either unaware of the [Swift SDK](https://github.com/facebook/facebook-swift-sdk) or choose to use the [Objective-C SDK](https://github.com/facebook/facebook-objc-sdk) instead.
+* There is confusion about the nature of the Swift SDK itself and no clear benefit to using it over the Objective-C version.
+* Any updates to the Objective-C SDK have to keep in mind that many of its users are coding in Swift.
+
+#### Requirements
+Keeping these facts in mind, we set out to design a strategy for modernizing that adheres to several broad requirements.
+
+* Must be able to add Swift specific interfaces (this is achieved by having the Swift SDK wrap existing classes)
+* Must deprecate Objective-C interfaces for Swift users when adding new Swift interfaces 
+* Must only deprecate Objective-C interfaces for Swift users. Objective-C users must not see deprecation warnings
+* Additive changes should be added to both SDKs when they are not constrained by language
+* Additions should be tested so that they offer a quality benefit on top of whatever other benefits they give
+
+#### Goals for modernization
+
+The ultimate goal of the Swift SDK is to provide modernizations that cannot be achieved in Objective-C such as:
+
+* Adding support for fetching `Codable` objects from the Graph API
+* Adding `Combine` publishers to existing services
+* Adding type-safety and validation to common types
+* Adding SwiftUI components for `LoginButton` and other UI components
+
+**Note:** Open an issue on Github or a pull request modifying this list if there is something else you'd like to see.
+
+
+## Simplified Example
+
+The pattern will be to add an interface that is specific to Swift while hiding the underlying Objective-C interface that it depends on.
+
+I will go into more detail but before that; a simplistic example. These would be the steps for wrapping method `x`:
+
+1. In the Swift SDK, add method `wrapsX`. It will wrap method `x` imported from the Objective-C SDK.
+2. In the Objective-C SDK add underscored prefix method `__x` that wraps method `x`.
+3. In the Objective-C SDK, tag method `x` with the `FB_SWIFT_DEPRECATED` macro shown below.
+    ```
+    #if __swift__
+	#define FB_SWIFT_DEPRECATED(message) __attribute__((deprecated(message)));
+	#else
+	#define FB_SWIFT_DEPRECATED(...)
+	#endif
+	```
+3. In the Swift SDK, update method `wrapsX` to call `__x`
+4. In the Objective-C SDK, replace the `FB_SWIFT_DEPRECATED` macro with the [`NS_REFINED_FOR_SWIFT`](https://developer.apple.com/documentation/swift/objective-c_and_c_code_customization/improving_objective-c_api_declarations_for_swift) macro.
+5. In the Objective-C SDK, delete method `__x` as it will be provided by `NS_REFINED_FOR_SWIFT`.
+
+If that all made perfect sense to you then feel free to skip the detailed example below.
+
+## Detailed Example
+
+Consider, for instance, a function in Objective-C that accepts a  pointer to a user.
```

Comment:
Will add the word object in addition to what's there. I think it's important to call out that it's a pointer because newer swift devs do not always draw that inference from the word 'object'.